staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -950,6 +951,13 @@ func (c *cacheWatcher) process(initEvents []*watchCacheEvent, resourceVersion ui
 		klog.V(2).Infof("processing %d initEvents of %s took %v", len(initEvents), objType, processingTime)
 	}
 
+	// With some events already sent, update resourceVersion so that
+	// events that were buffered and not yet processed won't be delivered
+	// to this watcher second time causing going back in time.
+	if len(initEvents) > 0 {
```

Comment:
slight preference for adjusting `watchRV` where we calculate initEvents, and passing a `watchRV` that doesn't require modification into newCacheWatcher

Reply:
Makes sense - done.